services/checks/funcs.go

```diff
@@ -80,3 +82,48 @@ func formatVersionNum(args ...interface{}) (interface{}, error) {
 	}
 	return p.String(), nil
 }
+
+func GetAdditionalContext() map[string]starlark.GoFunc {
+	return map[string]starlark.GoFunc{
+		"ip_is_private": ipIsPrivate,
+	}
+}
+
+var privateAddressBlocks = []string{
+	// private blocks, see https://tools.ietf.org/html/rfc1918
+	"10.0.0.0/8",
+	"172.16.0.0/12",
+	"192.168.0.0/16",
+	// link-local block, see https://tools.ietf.org/html/rfc3927
+	"169.254.0.0/16",
+	// loop-back block, see https://tools.ietf.org/html/rfc5735
+	"127.0.0.0/8",
+}
+
+// ipIsPrivate accepts a single string argument (IP address) and
+// returns true for a private address, otherwise false.
+func ipIsPrivate(args ...interface{}) (interface{}, error) {
+	if l := len(args); l != 1 {
+		return nil, errors.Errorf("expected 1 argument, got %d", l)
+	}
+
+	ip, ok := args[0].(string)
+	if !ok {
+		return nil, errors.Errorf("expected string argument, got %[1]T (%[1]v)", args[0])
+	}
+
+	ipAddress := net.ParseIP(ip)
+
+	for _, b := range privateAddressBlocks {
+		_, network, err := net.ParseCIDR(b)
+		if err != nil {
+			return nil, errors.Errorf("unable to parse CIDR: %s, reason: %s", b, err)
+		}
```

Comment:
Can we make it before that function invocation? I don't know, maybe even in `init` or somewhere else?

Reply:
is there a reason why you want it to be outside the function?

Reply:
To avoid parsing what is essentially constants (if Go had full contents support) on every function call. I'm with @artemgavrilov there. Let's define privateAddressBlocks as `[]*net.IPNet` and fill it in `init()`.